fclient/performjoin.go

```diff
@@ -0,0 +1,238 @@
+package fclient
```

Comment:
I don't like this file being in `fclient`. The point of `fclient` was to have dumb boring HTTP wrappers, of which this is not. Please move this to the root level and we will shift it to a `/pdu` package when we're ready.

This means you cannot use `FederationClient` else you'll get a circular import error. Instead, define a `JoinClient` interface with the subset of functions from `FederationClient` that you need (in this case, make join and send join).